randovania/layout/base/base_configuration.py

```diff
@@ -86,6 +88,11 @@ def dangerous_settings(self) -> list[str]:
         if self.first_progression_must_be_local:
             result.append("Requiring first progression to be local causes increased generation failure.")
 
+        if self.logical_pickup_placement is not LogicalPickupPlacementConfiguration.MINIMAL:
```

Comment:
The biggest concern is the `ALL` setting.
Also, can you try adding a `def dangerous_settings(self) -> list[str]:` method to `LogicalPickupPlacementConfiguration`? I think it might just work.